### PR TITLE
Added CLI option to delete a project

### PR DESCRIPTION
**Pull request by milanmayr**
I needed to disable and delete projects systematically, but I had no option to do this. (Later, I did find that you guys provided scripts for this stuff on your website). So, I extended the CLI to include the `delete-project` and `disable-projects` commands.

I was able to build the solution, and I tested the MacOS executable, successfully disabling and deleting a project.

### CODE REVIEW

File: source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public DisableProjectCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, IOctopusClientFactory clientFactory, ICommandOutputProvider commandOutputProvider)
+            : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider)
+        {
+            var options = Options.For("Project disablement");
+            options.Add<string>("name=", "The name of the project.", v => ProjectName = v);
+        }
+
+        public string ProjectName { get; set; }
+        public string ProjectGroupName { get; set; }
+        public bool ProjectExists { get; set; }
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs:26]:
```suggestion
```
Remove unused properties

File: source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        public DeleteProjectCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, IOctopusClientFactory clientFactory, ICommandOutputProvider commandOutputProvider)
+            : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider)
+        {
+            var options = Options.For("Project deletion");
+            options.Add<string>("name=", "The name of the project.", v => ProjectName = v);
+        }
+
+        public string ProjectName { get; set; }
+        public string ProjectGroupName { get; set; }
+        public bool ProjectExists { get; set; }
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs:27]:
```suggestion
```
Remove unused properties

File: source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        public void PrintJsonOutput()
+        {
+            commandOutputProvider.Json(new
+            {
+                Project = new
+                {
+                    project.Id,
+                    project.Name,
+                    ProjectDeleted = ProjectDeleted
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs:71]:
```suggestion
                    ProjectDeleted
```
remove redundant assignment

File: source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+            commandOutputProvider.Information("Finding project: {Project:l}", ProjectName);
+
+            project = await Repository.Projects.FindByName(ProjectName).ConfigureAwait(false);
+            if (project == null)
+            {
+                commandOutputProvider.Information("The project {Project:l} does not exist", project.Name);
+                return;
+
+                throw new CommandException($"The project {project.Name} does not exist.");
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs:41]:
```suggestion
                throw new CouldNotFindException("project");
```

File: source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            // Get project
+            project = await Repository.Projects.FindByName(ProjectName).ConfigureAwait(false);
+
+            // Check that is project exists and isn't already disabled
+            if (project == null)
+            {
+                commandOutputProvider.Information("The project {Project:l} does not exist.", project.Name);
+                return;
+
+                throw new CommandException($"The project {project.Name} does not exist.");
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs:43]:
```suggestion
                throw new CouldNotFindException("project");
```

File: source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            else if (project.IsDisabled == true)
+            {
+                commandOutputProvider.Information("The project {Project:l} is already disabled.", project.Name);
+                return;
+
+                throw new CommandException($"The project {project.Name} is already disabled.");
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DisableProjectCommand.cs:50]:
```suggestion
```

File: source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+            commandOutputProvider.Information("Deleting project: {Project:l}", ProjectName);
+
+            try
+            {
+                await Repository.Projects.Delete(project).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                ProjectDeleted = false;
```
Comment (slewis74) [inline source/Octopus.Cli/Commands/Project/DeleteProjectCommand.cs:52]:
```suggestion
```

**Review by slewis74 — CHANGES_REQUESTED** [review]
Thank you for your contribution and apologies for the delay in getting to the review. Overall this is looks fine. There are a couple of minor cleanup things and one adjustment to the exception control flow that we'd request.

**Comment by milanmayr** [comment]
@slewis74 thanks for the review! Your suggestions were really helpful, and I incorporated all of them.

I also imported a namespace so those new Exception types would be recognized.

**Review by slewis74 — APPROVED** [review]
👍

**Comment by slewis74** [comment]
Just a heads up, we've been doing some renovations to the build/deploy pipeline for the CLI over the last week or so. There was a glitch with `9.0.0` but these changes have been shipped in `9.1.1`.

**Pull request merged**